experimental/handoff-library/csharp_dotnetcore/Microsoft.Bot.Builder.Handoff/HandoffHttpSupport.cs

```diff
@@ -0,0 +1,235 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.Schema;
+using Microsoft.Bot.Schema.Handoff;
+using Microsoft.Rest;
+using Newtonsoft.Json;
+
+// This file mimics the generated code in Conversations.cs which disables the following SA warnings:
+#pragma warning disable SA1629
+#pragma warning disable SA1513
+#pragma warning disable SA1515
+#pragma warning disable SA1312
+#pragma warning disable SA1122
+#pragma warning disable SA1507
+#pragma warning disable SA1508
+
+namespace Microsoft.Bot.Builder.Handoff
+{
+    /// <summary>
+    /// Making HTTP calls for handoff. When handoff is integrated
+    /// into the SDK proper, this code migrates to Conversations.cs.
+    /// </summary>
+    public static class HandoffHttpSupport
+    {
+        public static async Task<HttpOperationResponse<ResourceResponse>> HandoffWithHttpMessagesAsync(IServiceOperations<ConnectorClient> conversations, string conversationId, HandoffParameters handoffParameters, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (conversationId == null)
+            {
+                throw new ValidationException(ValidationRules.CannotBeNull, "conversationId");
+            }
+            if (handoffParameters == null)
+            {
+                throw new ValidationException(ValidationRules.CannotBeNull, "handoffParameters");
+            }
+
+            // Construct URL
+            var _baseUrl = conversations.Client.BaseUri.AbsoluteUri;
+            var _url = new System.Uri(new System.Uri(_baseUrl + (_baseUrl.EndsWith("/") ? "" : "/")), "v3/conversations/{conversationId}/handoff").ToString();
+            _url = _url.Replace("{conversationId}", System.Uri.EscapeDataString(conversationId));
+            // Create HTTP transport objects
```

Comment:
A few things here. :) 

1. Please don't use the "_" for local stack variables. Our coding standard is to use "_" for private member variables. 
2. This code is complex, and needs a comment or two. 
3. The Template should be moved to a constant with a comment. 

Why is  variable called "_url" being constructed from 2 URI's, and then converted to a String? Consider a more descriptive / accurate name, and consider keeping it a URI until it's needed as a string. 

(for example, down on line 52, you re-convert it to a URI).

Reply:
This file was generated from the tool, just like Conversations.cs. I rewrote it completely and addressed all comments.